### PR TITLE
Rework X11 backend to use EGL for finding gbm device and associated commits

### PR DESCRIPTION
**Pull request by i509VCB**

### CODE REVIEW

**Comment by i509VCB** [comment]
I have no earthly idea why thiserror freaks out on github actions but compiles locally all fine.

File: build.rs [hunk 1]
```diff
@@ -17,13 +17,20 @@ fn gl_generate() {
                 "EGL_EXT_create_context_robustness",
                 "EGL_KHR_create_context_no_error",
                 "EGL_KHR_no_config_context",
+                "EGL_EXT_device_base",
+                "EGL_EXT_device_enumeration",
+                "EGL_EXT_device_query",
+                "EGL_EXT_device_drm",
+                "EGL_KHR_stream",
+                "EGL_KHR_stream_producer_eglsurface",
```
Comment (Drakulix) [inline build.rs:25]: I do not think we need the last two for this feature?

**Review by Drakulix — APPROVED** [review]
Looks good to me!

File: build.rs (hunk 1 above)
Comment (Drakulix) [reply build.rs:25]: Oh right, that was me. Probably an oversight when copying parts of the `EGLDevice` code from my eglstream implementation.

**Pull request merged**